### PR TITLE
Add Fusion contract

### PR DESCRIPTION
**Pull request by tomjohn1028**
* copy over the fusion/crafting contract
* change contract name to fusion

### CODE REVIEW

File: programs/fusion/src/instructions/craft.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        &"crafting".as_bytes(),
+        &formula.to_account_info().key.to_bytes()[..32],
+        &[bump],
+    ];
+    let signer = &[&seeds[..]];
+
+    for item in formula.output_items.iter() {
+        // handle case where the output Item is a master edition
+        if item.is_master_edition {
+            msg!("Item is a master edition! Print that mother F@#!$");
```
Comment (VadimGrozinok) [inline programs/fusion/src/instructions/craft.rs:86]: are you sure that we want to leave it as it is ? 😁

**Review by VadimGrozinok — COMMENTED** [review]
Looks good, I like it
Regarding the directories organisation I think it should looks like this:
- change path to src from `program/fusion/src` to `fusion/src` like in nft-packs program
- Fusion readme move to fusion program dir
- update main Readme in repo
- tests for Fusion program also move to Fusion program dir

@austbot what do you think ?

File: programs/fusion/src/instructions/craft.rs (hunk 1 above)
Comment (tomjohn1028) [reply programs/fusion/src/instructions/craft.rs]: Lolll removed the messages

**Comment by tomjohn1028** [comment]
> he was just following anchor semantics

Indeed

**Review by VadimGrozinok — APPROVED** [review]

**Pull request merged**